tests/test_vtk_multistage_bug.py

```diff
@@ -0,0 +1,198 @@
+                                       max_delta_time_factor=1000)
+    convergence_criterion = DisplacementConvergenceCriteria(displacement_relative_tolerance=1.0E-12,
+                                                            displacement_absolute_tolerance=1.0E-6)
+    stress_initialisation_type = StressInitialisationType.NONE
+    solver_settings = SolverSettings(analysis_type=analysis_type,
+                                     solution_type=solution_type,
+                                     stress_initialisation_type=stress_initialisation_type,
+                                     time_integration=time_integration,
+                                     is_stiffness_matrix_constant=False,
+                                     are_mass_and_damping_constant=False,
+                                     convergence_criteria=convergence_criterion,
+                                     rayleigh_k=0.0,
+                                     rayleigh_m=0.0)
+
+    # Set up problem data
+    problem = Problem(problem_name="test_multi_stage_block", number_of_threads=2, settings=solver_settings)
+    model_stage_1.project_parameters = problem
+
+    # Define the results to be written to the output file
+    # Nodal results
+    nodal_results = [NodalOutput.DISPLACEMENT, NodalOutput.VELOCITY]
+
+    # Define the output process
+    model_stage_1.add_output_settings(output_parameters=VtkOutputParameters(file_format="ascii",
+                                                                            output_interval=output_interval,
+                                                                            nodal_results=nodal_results,
+                                                                            gauss_point_results=[],
+                                                                            output_control_type="step"),
+                                      output_dir="output",
+                                      output_name="vtk_output")
+
+    # define the STEM instance
+    input_folder = "tests/inputs_kratos"
+    stem = Stem(model_stage_1, input_folder)
+
+    # create new stage
+    model_stage_2 = stem.create_new_stage(delta_time_stage_2, 0.05)
+
+    # Set up solver settings for the new stage
+    model_stage_2.project_parameters.settings.solution_type = SolutionType.DYNAMIC
+    model_stage_2.project_parameters.settings.rayleigh_k = 1e-6
+    model_stage_2.project_parameters.settings.rayleigh_m = 0.02
+
+    model_stage_2.process_model_parts[0].parameters.value = [0, -1050, 0]
+    model_stage_2.output_settings[0].output_parameters.nodal_results = [NodalOutput.DISPLACEMENT, NodalOutput.VELOCITY]
+
+    # add the new stage to the calculation
+    stem.add_calculation_stage(model_stage_2)
+
+    # write the kratos input files
+    stem.write_all_input_files()
+
+    # Run Kratos calculation
+    # --------------------------------
+    stem.run_calculation()
+
+
+class TestVtkMultistageBug:
```

Comment:
missing test 3 stage calculation where output interval is greater than the first 2 stages

Reply:
missing test for the case where output interval is greater than the number of timesteps in stage 2 but not stage 1